applications/convection_diffusion_application/python_scripts/convection_diffusion_base_solver.py

```
@@ -0,0 +1,755 @@
+        convention_diffusion_settings = KratosMultiphysics.ConvectionDiffusionSettings()
+        density_variable = self.settings["convection_diffusion_variables"]["density_variable"].GetString()
+        if (density_variable is not ""):
+            convention_diffusion_settings.SetDensityVariable(KratosMultiphysics.KratosGlobals.GetVariable(density_variable))
+        diffusion_variable = self.settings["convection_diffusion_variables"]["diffusion_variable"].GetString()
+        if (diffusion_variable is not ""):
+            convention_diffusion_settings.SetDiffusionVariable(KratosMultiphysics.KratosGlobals.GetVariable(diffusion_variable))
+        unknown_variable = self.settings["convection_diffusion_variables"]["unknown_variable"].GetString()
+        if (unknown_variable is not ""):
+            convention_diffusion_settings.SetUnknownVariable(KratosMultiphysics.KratosGlobals.GetVariable(unknown_variable))
+        volume_source_variable = self.settings["convection_diffusion_variables"]["volume_source_variable"].GetString()
+        if (volume_source_variable is not ""):
+            convention_diffusion_settings.SetVolumeSourceVariable(KratosMultiphysics.KratosGlobals.GetVariable(volume_source_variable))
+        surface_source_variable = self.settings["convection_diffusion_variables"]["surface_source_variable"].GetString()
+        if (surface_source_variable is not ""):
+            convention_diffusion_settings.SetSurfaceSourceVariable(KratosMultiphysics.KratosGlobals.GetVariable(surface_source_variable))
+        projection_variable = self.settings["convection_diffusion_variables"]["projection_variable"].GetString()
+        if (projection_variable is not ""):
+            convention_diffusion_settings.SetProjectionVariable(KratosMultiphysics.KratosGlobals.GetVariable(projection_variable))
+        convection_variable = self.settings["convection_diffusion_variables"]["convection_variable"].GetString()
+        if (convection_variable is not ""):
```

else raise no?

---

what?

---

If the var name is empty you should raise an exception no?

---

Nope, it is already overrided in the init, but in case you are interested in not set that variable you can use "" instead